### PR TITLE
feat(words): add pandas related words

### PR DESCRIPTION
**Pull request by bopeng-saitama**
Signed-off-by: Bo Peng <82485734+bopeng-saitama@users.noreply.github.com>

This PR adds pandas related words.

Attributes list:

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iloc.html?highlight=iloc#pandas.DataFrame.iloc
https://pandas.pydata.org/docs/reference/api/pandas.isna.html?highlight=isna#pandas.isna
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html?highlight=dataframe#pandas.DataFrame
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.itertuples.html?highlight=itertuples#pandas.DataFrame.itertuples
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dtypes.html?highlight=dtypes#pandas.DataFrame.dtypes

### CODE REVIEW

**Review by takam5f2 — COMMENTED** [review]
@bopeng-saitama Thank you for making PR.

@kenji-miyake I think it is okay to merge this PR to main branch. Could you tell me if there is something wrong?

File: .cspell.json [hunk 1]
```diff
@@ -444,6 +444,7 @@
     "datadir",
     "dataend",
     "datafiles",
+    "dataframe",
```
Comment (kenji-miyake) [inline .cspell.json:447]: I think this is usually written as `DataFrame`, which doesn't cause an error.

```suggestion
```

**Review by kenji-miyake — CHANGES_REQUESTED** [review]

File: .cspell.json (hunk 1 above)
Comment (takam5f2) [reply .cspell.json:447]: Sorry, but I failed in checking it. This must be excluded in this PR.

@bopeng-saitama  Could you replace dataframe by another commonly-used word in CARET's source code.

Comment (bopeng-saitama) [reply .cspell.json]: @takam5f2 Noted on this. I'll do it right away.

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**